app/src/main/java/com/example/coronadiagnosticapp/ui/activities/OxymeterActivity.java

```diff
@@ -137,50 +129,29 @@ public void onPreviewFrame(byte[] data, Camera cam) {
             double totalTimeInSecs = (endTime - startTime) / 1000d; //to convert time to seconds
             if (totalTimeInSecs >= 30 && startTime != 0) { //when 30 seconds of measuring passes do the following " we chose 30 seconds to take half sample since 60 seconds is normally a full sample of the heart beat
                 SamplingFreq = (counter / totalTimeInSecs);
-                ArrayList<Double> RedMoveAverage = calculateMovingAverage();
-//--------------------------CSV Writing-------------------------------
-                writeCSV(rows);
-//--------------------------------------------------------------------
+                ArrayList<Double> RedMoveAverage = calculateMovingAverage(RedAvgList);
                 ArrayList<Integer> peaksList = createWindowsToFindPeaks(RedMoveAverage);
-                FindIntervalsAndCalculateBPM(peaksList);
-
-                Double[] Red = RedAvgList.toArray(new Double[RedAvgList.size()]);
-                Double[] Blue = BlueAvgList.toArray(new Double[BlueAvgList.size()]);
-                Double[] Green = GreenAvgList.toArray(new Double[GreenAvgList.size()]);
-
-                // double HRFreq = Fft.FFT(Red, counter, SamplingFreq);
-                // double bpm = (int) ceil(HRFreq * 60);
-
-
-                o2 = (int) calculateSPO2(Red,Blue);
-                Breath = (int) calculateAverageFourierBreathAndBPM(Red,Green,Blue)[0]; // 0 stands for breath respiration value
-                Beats = (int) calculateAverageFourierBreathAndBPM(Red,Green,Blue)[1]; // 0 stands for Heart Rate value
-
-            }
-
+                double peakBpm = FindIntervalsAndCalculateBPM(peaksList);
+                //--------------------------CSV Writing-------------------------------
+                writeCSV(rows);
+                //--------------------------------------------------------------------
+                int o2 = (int) calculateSPO2(RedAvgList, BlueAvgList);
+                double[] breathAndBPM = calculateAverageFourierBreathAndBPM(RedAvgList, GreenAvgList);
+                int Breath = (int) breathAndBPM[0]; // 0 stands for breath respiration value
+                int Beats = (int) breathAndBPM[1]; // 1 stands for Heart Rate value
 
-            if ((Beats != 0) && (o2 != 0) && (Breath != 0) && (peakBpm != 0)) {
```

Comment:
NOTE : i removed this condition since from what i understand this should happen right after the above calculation happen. 
The problem i'm not sure of is that `calculateAverageFourierBreathAndBPM` may return `0,0` in one of its flow. Is this flow reachable? because  if so removing this `if` will change the flow (and make the below code run). Is that the intended flow or should i change this ?

Reply:
UPDATE: just realized im stupid and there is no case where it returns `0,0`

Reply:
UPDATE: the commit `handle wired 0,0 state` should hanndle this now